cylc/flow/task_remote_mgr.py

```diff
@@ -139,23 +146,21 @@ def subshell_eval_reset(self):
             if value is not None:
                 del self.remote_command_map[key]
 
-    def remote_init(self, platform_name, curve_auth, client_pub_key_dir):
+    def remote_init(self, platform, curve_auth,
+                    client_pub_key_dir):
         """Initialise a remote [owner@]host if necessary.
 
-        Create UUID file on a platform ".service/uuid" for remotes to identify
-        shared file system with suite host.
-
         Call "cylc remote-init" to install suite items to remote:
             ".service/contact": For TCP task communication
             "python/": if source exists
 
         Args:
-            platform_name (str):
-                The name of the platform to be initialized.
             curve_auth (ThreadAuthenticator):
                 The ZMQ authenticator.
             client_pub_key_dir (str):
                 Client public key directory, used by the ZMQ authenticator.
+            platform (dict):
+                A dictionary containing information about platform.
```

Comment:
This seems to be an infinitely recursive definition :grin: `foo: a dictionary of information about foo`

Reply:
Yes, mind-bending - I'll change the description so it is a bit more informative!